src/components/errorBoundary/ErrorBoundary.tsx

```diff
@@ -0,0 +1,32 @@
+import React, { Component } from 'react';
+
+import { Oops } from 'components';
+
+type ErrorState = { error: Error | null, errorInfo: React.ErrorInfo | null }
+
+class ErrorBoundary extends Component {
+  state: ErrorState = {
+    error: null,
+    errorInfo: null,
+  }
+
+  componentDidCatch(error: Error, errorInfo: React.ErrorInfo): void {
+    this.setState({
+      error,
+      errorInfo,
+    });
+  }
```

Comment:
Из [доки](https://ru.reactjs.org/docs/react-component.html#static-getderivedstatefromerror):
В случае ошибки вы можете рендерить запасной интерфейс с помощью componentDidCatch(), вызвав setState. Однако, этот способ скоро будет считаться устаревшим. Используйте static getDerivedStateFromError() для рендера резервного интерфейса. Тут не так много переписыватаь, давай вот так сделаем

```suggestion
  static getDerivedStateFromError(error) {
    return { hasError: true };
  }
```

`componentDidCatch` можно оставить на будущее, но пустой, там можно будет настроить логирование

Reply:
Спасибо! Я читала доку, но что-то видимо пропустила этот момент. И было не оч понятно, зачем нам оба метода. Теперь все встало на свои места, спасибо